pytorch_lightning/overrides/base.py

```diff
@@ -54,19 +54,18 @@ def forward(self, *inputs, **kwargs):
             # ddp_plugin ``post_training_step`` hook
             if not self.module.automatic_optimization:
                 self.module.trainer.model.require_backward_grad_sync = False
-            warn_if_output_is_none(output, "training_step")
+
+            if self.module.automatic_optimization:
+                warn_if_output_is_none(output, "training_step")
 
         elif running_stage == RunningStage.TESTING:
             output = self.module.test_step(*inputs, **kwargs)
-            warn_if_output_is_none(output, "test_step")
 
         elif running_stage == RunningStage.EVALUATING:
             output = self.module.validation_step(*inputs, **kwargs)
-            warn_if_output_is_none(output, "validation_step")
 
         elif running_stage == RunningStage.PREDICTING:
             output = self.module.predict(*inputs, **kwargs)
```

Comment:
We should warn during predict call as we expect the user to return the predictions.

Reply:
added.